python/lsst/pipe/tasks/extended_psf.py

```diff
@@ -239,3 +245,250 @@ def readFits(cls, filename):
         """Alias for ``readFits``; exists for compatibility with the Butler.
         """
         return cls.read_fits(filename)
+
+
+class StackBrightStarsConfig(pexConfig.Config):
+    """Configuration parameters for StackBrightStarsTask.
+    """
+    subregion_size = pexConfig.ListField(
+        dtype=int,
+        doc="Size, in pixels, of the subregions over which the stacking will be "
+            "iteratively performed.",
+        default=(100, 100)
+    )
+    stacking_statistic = pexConfig.ChoiceField(
+        dtype=str,
+        doc="Type of statistic to use for stacking.",
+        default="MEANCLIP",
+        allowed={
+            "MEAN": "mean",
+            "MEDIAN": "median",
+            "MEANCLIP": "clipped mean",
+        }
+    )
+    num_sigma_clip = pexConfig.Field(
+        dtype=float,
+        doc="Sigma for outlier rejection; ignored if stacking_statistic != 'MEANCLIP'.",
+        default=4
+    )
+    num_iter = pexConfig.Field(
+        dtype=int,
+        doc="Number of iterations of outlier rejection; ignored if atackingStatistic != 'MEANCLIP'.",
+        default=3
+    )
+    bad_mask_planes = pexConfig.ListField(
+        dtype=str,
+        doc="Mask planes that, if set, lead to associated pixels not being included in the stacking of the "
+            "bright star stamps.",
+        default=('BAD', 'CR', 'CROSSTALK', 'EDGE', 'NO_DATA', 'SAT', 'SUSPECT', 'UNMASKEDNAN')
+    )
+    do_mag_cut = pexConfig.Field(
+        dtype=bool,
+        doc="Apply magnitude cut before stacking?",
+        default=False
+    )
+    mag_limit = pexConfig.Field(
+        dtype=float,
+        doc="Magnitude limit, in Gaia G; all stars brighter than this value will be stacked",
+        default=18
+    )
+
+
+class StackBrightStarsTask(pipeBase.CmdLineTask):
+    """Stack bright stars together to build an extended PSF model.
+    """
+    ConfigClass = StackBrightStarsConfig
+    _DefaultName = "stack_bright_stars"
+
+    def __init__(self, initInputs=None, *args, **kwargs):
+        pipeBase.CmdLineTask.__init__(self, *args, **kwargs)
+
+    def _set_up_stacking(self, example_stamp):
+        """Configure stacking statistic and control from config fields.
+        """
+        stats_control = afwMath.StatisticsControl()
+        stats_control.setNumSigmaClip(self.config.num_sigma_clip)
+        stats_control.setNumIter(self.config.num_iter)
+        bad_masks = self.config.bad_mask_planes
+        if bad_masks:
```

Comment:
Perhaps personal preference, but this could be combined into:
`if bad_masks := self.config.bad_mask_planes:`